### PR TITLE
Move all ONNX test dependencies to Docker

### PR DESCRIPTION
**Pull request by huydhn**
Per title.  This is the first one of a two-part process:

[x] Move all ONNX test dependencies to Docker https://github.com/pytorch/pytorch/pull/96590
[ ] Move the test model used by [TestFxToOnnxWithOnnxRuntime.test_gpt2_tiny](https://hud.pytorch.org/failure/FAILED%20test%2Fonnx%2Ftest_fx_to_onnx_with_onnxruntime.py%3A%3ATestFxToOnnxWithOnnxRuntime%3A%3Atest_large_scale_exporter_with_tiny_gpt2%20-%20requests.exceptions.ReadTimeout%3A%20HTTPSConnectionPool(host%3D'huggingface.co'%2C%20port%3D443)%3A%20Read%20timed%20out.%20(read%20timeout%3D10.0))

### CODE REVIEW

File: .ci/onnx/test.sh [hunk 1]
```diff
@@ -3,70 +3,11 @@
...
-# If pip is installed as root, we must use sudo.
-# CircleCI docker images could install conda as jenkins user, or use the OS's python package.
-PIP=$(which pip)
-PIP_USER=$(stat --format '%U' $PIP)
-CURRENT_USER=$(id -u -n)
-if [[ "$PIP_USER" = root && "$CURRENT_USER" != root ]]; then
-  MAYBE_SUDO=sudo
-fi
-
-# Uninstall pre-installed hypothesis and coverage to use an older version as newer
```
Comment (ZainRizvi) [inline .ci/onnx/test.sh:46]: Same Q about hypothesis

File: .ci/docker/common/install_onnx.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+set -ex
+
+source "$(dirname "${BASH_SOURCE[0]}")/common_utils.sh"
+
+# A bunch of custom pip dependencies for ONNX
+pip_install beartype==0.10.4 \
+  filelock==3.9.0 \
```
Comment (ZainRizvi) [inline .ci/docker/common/install_onnx.sh:9]: Readability nit

```suggestion
pip_install \
  beartype==0.10.4 \
  filelock==3.9.0 \
```

File: scripts/onnx/test.sh [hunk 3]
```diff
@@ -24,15 +24,6 @@ do
...
 
-if [[ $PARALLEL == 1 ]]; then
-    pip install pytest-xdist
-fi
-
-# pytest, scipy, hypothesis: these may not be necessary
-# pytest-cov: installing since `coverage run -m pytest ..` doesn't work
-# pytest-subtests: unittest subtests support for pytest
-# parameterized: parameterizing test class
-pip install pytest scipy hypothesis pytest-cov pytest-subtests parameterized
```
Comment (ZainRizvi) [inline scripts/onnx/test.sh:35]: Not all of these have been moved over. Where they already installed elsewhere?

**Review by ZainRizvi — APPROVED** [review]
Awesome to see all these files getting dockerized!

File: .ci/onnx/test.sh (hunk 1 above)
Comment (huydhn) [reply .ci/onnx/test.sh:46]: Like click, hypothesis has already been installed by Docker `requirements-ci.txt`.  I would love to have all CI pip dependencies in that file eventually (we're close I think)

File: .ci/onnx/test.sh [hunk 4]
```diff
@@ -3,70 +3,11 @@
...
-fi
-# These additional packages are needed for circleci ROCm builds.
-if [[ $BUILD_ENVIRONMENT == *rocm* ]]; then
-    # Need networkx 2.0 because bellmand_ford was moved in 2.1 . Scikit-image by
-    # defaults installs the most recent networkx version, so we install this lower
-    # version explicitly before scikit-image pulls it in as a dependency
-    pip install networkx==2.0
-    # click - onnx
-    pip install --progress-bar off click protobuf tabulate virtualenv mock typing-extensions
-fi
```
[the remark this answers is not in the document]
Comment (huydhn) [reply .ci/onnx/test.sh:28]: Yeah, the script is only run for onnx.  There is no way the build environment could be both onnx and rocm or android.  They are dead code.

File: .ci/docker/common/install_onnx.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
```
Comment (titaiwangms) [inline .ci/docker/common/install_onnx.sh:1]: Can ONNX exporter team have the access to change this file?https://github.com/pytorch/pytorch/blob/e7d795dccd28bfa556ee5b073520c3e00a68aabe/.github/merge_rules.yaml#L1-L4

Comment (huydhn) [reply .ci/docker/common/install_onnx.sh:1]: Good point!  I will update the merge rules accordingly.  Thank you for noticing this.

Comment (titaiwangms) [reply .ci/docker/common/install_onnx.sh:1]: Thanks!

**Pull request closed without merging**